### ISSUE TITLE
Injected HttpClient fails in test if rebuildContext is set to true (with reproducer)

### ISSUE THREAD

**Issue by oscarfh** [micronaut-projects/micronaut-core#5453]
I faced this issue during development and I was able to extract to a reproducer that can be seen here:
https://github.com/oscarfh/micronaut-test-error

Simply check it our and execute the DemoTest test.
There you will see two identical tests, if executed together, the first passes, the second always fails. If executed separately, both pass.

There is no much magic there, injecting an httpclient and calling the controller that exist in the application. The issue seems to be the `rebuildContext` flag that is set on line 25. (Remove that and you will get green tests).

This feel related to this issue I once found (and was fixed): https://github.com/micronaut-projects/micronaut-core/issues/2138

**Comment by graemerocher** [comment]
I don't believe there is much we can do about this `rebuildContext` is incompatible with `@TestInstance(TestInstance.Lifecycle.PER_CLASS)` as you have on your test since it prevents re-injecting all dependencies for each test

**Comment by oscarfh** [comment]
@graemerocher The problem persists even after removing `@TestInstance(TestInstance.Lifecycle.PER_CLASS)`.

**Comment by nayavu** [comment]
I've got the same issue. 

As an ugly workaround, I do the following:
```kotlin
@MicronautTest(rebuildContext = true)
class Test {
    @field:Inject
    lateinit var embeddedServer: EmbeddedServer

    @field:Inject
    @field:Client("/")
    lateinit var client: RxHttpClient

    @BeforeEach
    fun initialise() {
        if (firstRun) {
            firstRun = false
        } else {
            embeddedServer.refresh()
        }
    }

   // tests 

    companion object {
        @JvmStatic
        var firstRun: Boolean = true
    }
}
```

**Comment by ojebari-ma** [comment]
Can you please confirm that this is still an issue in the latest version of Micronaut?
